src/app/shell/personal-cabinet/personal-cabinet.component.ts

```diff
@@ -46,6 +47,7 @@ export class PersonalCabinetComponent implements OnInit, OnDestroy {
         })
       )
     );
+    this.subrole$.subscribe((subrole: string) => this.subrole = subrole)
```

Comment:
maybe you can remove subscribe here and use async in the template? or add takeUntil to the subscription